store/auth.go

```diff
@@ -159,3 +161,24 @@ func DischargeAuthCaveat(username, password, macaroon, otp string) (string, erro
 	}
 	return responseData.Macaroon, nil
 }
+
+func decodeUnauthorizedError(body io.ReadCloser) error {
+	// first check if the server even thinks we tried to authenticate
+	b, _ := ioutil.ReadAll(body)
+	if string(b) == "Authorization Required" {
+		return ErrNeedsAuthorization
+	}
+
+	// check json details
+	var msg authError
+	dec := json.NewDecoder(body)
+	if err := dec.Decode(&msg); err != nil {
```

Comment:
Seems strange to do that when we don't quite know if it's a JSON message (code above assumes it's not).

This error will return an ugly and incomprehensible message to the user.